build_tools/python/e2e_test_framework/definitions/iree_definitions.py

```diff
@@ -166,10 +167,12 @@ def __str__(self):
   def materialize_import_flags(self,
                                model: common_definitions.Model) -> List[str]:
     """Materialize flags with dependent values."""
-    return [
-        flag.replace(IMPORT_CONFIG_ENTRY_FUNCTION_PLACEHOLDER,
-                     model.entry_function) for flag in self.import_flags
-    ]
+    return utils.transform_flags(
+        flags=self.import_flags,
+        map_funcs=[
+            lambda value: value.replace(
+                IMPORT_CONFIG_ENTRY_FUNCTION_PLACEHOLDER, model.entry_function)
```

Comment:
Could this be template strings? https://docs.python.org/3/library/string.html#template-strings

Reply:
I also think this will better handle multiple substitutions (instead of calling replace for each variable)

Reply:
In general I want to simply use replace/template across the board. But the logic to substitute path and convert to platform-dependent path is too complicated for them.

I was thinking about avoiding path in the flags (e.g. `--iree-scheduling-dump-statistics-file=scheduling_stats.json` then we change the working directory when running the compile commands), but that looks more complicated to me in other parts

Reply:
Variable substitution is probably the most common use case. Added a sugar function `substitute_flag_vars` based on `transform_flags`